From aadc5f904000dc742f2b5623ba876a5dd800ccf2 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 19 Jun 2024 15:55:44 +0300
Subject: [PATCH] [SW-185990] gaudi3: write comp version to descriptor

Write SPI component version to it's descriptor.
Only executable components have versions at the moment. Versions
for other non-executable components can be added later.

Change-Id: I17bd6aa2ac711fb0ee465bf467d74b951da64dfa
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 7dce3b8..63f6eb6 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -375,7 +375,26 @@
 }
 
 function get_spi_component_ver() {
-	echo "this is a dummy component version"
+	local img_name=${1^^}
+	local baremetal_comp=("PPBOOT" "PIF_ICCM" "PIF_DCCM" "EROM_PCIE" "EROM_RECOVERY")
+	local preboot_comp=("PREBOOT")
+	local ver
+
+	if [ $img_name == $preboot_comp ]; then
+		ver=$(cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/preboot-custom/localversion-habanalabs)
+		echo ${ver#*-}
+		return
+	fi
+
+	for i in "${baremetal_comp[@]}"; do
+		if [ $img_name == $i ]; then
+			ver=$(cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/baremetal-custom/localversion-habanalabs)
+			echo ${ver#*-}
+			return
+		fi
+	done
+
+	echo "no version"
 }
 
 function print_image_version {
